web-app/src/main/java/com/infoshare/webapp/service/FileJSONService.java

```diff
@@ -0,0 +1,37 @@
+package com.infoshare.webapp.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.infoshare.webapp.model.Questions;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class FileJSONService {
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private List<Questions> questions;
```

Comment:
```suggestion
        private final ObjectMapper objectMapper;
    private List<Questions> questions;

    public FileJSONService(ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }
```

Reply:
poprawiłem, chociaż nie do końca rozumiem potrzebę wstrzykiwania?

Reply:
Słowo kluczowe "new" służy do tworzenia nowych obiektów w pamięci, które można później wykorzystać w programie. W ten sposób możesz tworzyć obiekty i korzystać z nich bezpośrednio wewnątrz kodu.

Z drugiej strony, wstrzykiwanie zależności jest techniką programowania, która polega na wstrzykiwaniu gotowych obiektów do innych klas, w których są one używane. W ten sposób nie tworzysz nowych obiektów, ale wykorzystujesz istniejące już obiekty i wstrzykujesz je tam, gdzie są potrzebne.

Wstrzykiwanie zależności ma wiele korzyści, takich jak poprawa modularności kodu, łatwiejsze testowanie, zmniejszenie liczby zależności między klasami oraz poprawa czytelności kodu. Pozwala również na łatwe podmienianie konkretnych implementacji interfejsów lub klas podczas działania programu, co ułatwia modyfikowanie i rozwijanie aplikacji.

W przypadku tworzenia obiektów przez słowo kluczowe "new", klasa odpowiedzialna za tworzenie obiektu zna wszystkie zależności i decyduje, jakie obiekty należy utworzyć. W przypadku wstrzykiwania zależności, to klasa korzystająca z obiektu określa, jakie obiekty są potrzebne do jej działania, a kontener wstrzykiwania zależności dostarcza te obiekty.